src/index.js

```diff
@@ -299,15 +297,31 @@ export class CancelToken {
         resolve(reason)
       }
 
-      // notify sync listeners (for race)
-      const listeners = this._listeners
-      if (listeners !== null) {
-        this._listeners = null
-        _forArray(listeners, listener => void listener(reason))
+      const handlers = this._handlers
+      if (handlers !== null) {
+        this._handlers = null
+
+        const { promise, resolve } = defer()
```

Comment:
Because I choose to not return a promise when the cancelation is instantaneous (no async handlers).

I have done so to not break sync `CancelToken.race()`.